src/simulation.jl

```diff
@@ -183,6 +183,67 @@ function simulate(rng::AbstractRNG, simulation::Simulation; return_epoched::Bool
 
 end
 
+"""TODO docstring
+s: Vector of AbstractContinuousSignal and/or AbstractNoise 
+NOTE:
+1. It is assumed that `AbstractContinuousSignal.controlsignal` 
+    is defined from the start of the simulation. If you require the artifact to start at
+    a different time, please take care to pre-pad the controlsignal with zeros.
+2. EEG simulation components must be multichannel, and it is assumed that
+    the number and position of EEG channels are the same as in the artifact head model. 
+"""
+function simulate(rng::AbstractRNG,d::AbstractDesign,c::AbstractComponent,o::AbstractOnset,s::AbstractVector)
+    @assert all(x -> x isa AbstractContinuousSignal || x isa AbstractNoise, s) "Artifact-related inputs should all be of type AbstractContinuousSignal or AbstractNoise"
+    sim = Simulation(d, c, o, NoNoise()) # since generated controlsignal might depend on some aspect of the simulation
+
+    # Current Assumptions:
+    # events from artifacts - not considered/handled right now
+    # EEG component is assumed to be multichannel and having the same number and positions of channels as the artifact.
+    # noise is considered to be independent of EEG/artifacts.
+    # controlsignals are generated only for AbstractContinuousSignal and not AbstractNoise.
+    # currently any number of noise components and/or artifacts are allowed. e.g. we can simulate RedNoise as well as PinkNoise and multiple EyeMovements each starting from t=0.
+    # eeg and artifact signals are simulated separately and then added together, padding to the larger of the two. Then noise is added to this signal.
+    # multiple EyeMovement artifacts are currently allowed. (TODO: check if it makes sense to restrict this. This could be useful if different Eyemovements have different offsets?)
+    # PLN simulation assumes that the sampling frequency of the final signal is the same as the sampling frequency of the other signals (eeg/artifacts). TODO: declare this more explicitly in the docstring.
+
+    println("Simulating EEG with no noise...")
+    eeg_signal,evts = simulate(rng,d,c,o,NoNoise());
+
+    sim_artifacts = [x for x in s if !(x isa PowerLineNoise)] # ignore PLN for now, it is handled separately since it needs to know the length of the final signal
```

Comment:
I handled such cases in the past, that a `length(x::AbstractContinuousSignal)` needs to be defined (e.g. for `AbstractComponents` - could this be a solution? I dont want any ContinuousSignal be treated "special"